.github/workflows/build-test.yml

```diff
@@ -1163,3 +1163,25 @@ jobs:
         github-token: ${{ secrets.GITHUB_TOKEN }}
         comment-on-alert: true
         alert-comment-cc-users: '@gnrunge,@sffc,@zbraniecki,@echeran'
+
+  # Notify on slack  
+  notify-slack:
+    ## For Rob's PR:
+    ## Skipped tasks: bench-perf, bench-memory, bench-binsize, bench-datasize
+    ## needs: [msrv, test, testdata, test-docs, full-datagen, ffi, verify-ffi, gn, wasm, fmt, tidy, clippy, doc]
+    
+    # Skipped tasks: benchmark, memory, datasize, binsize
+    needs: [check-msrv, test, testdata, test-docs, full-datagen, ffi, verify-ffi, gn, features, wasm, fmt, tidy, clippy, doc, ]
+    if: ${{ always() && contains(needs.*.result, 'failure') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Notify slack of failure
+        uses: ravsamhq/notify-slack-action@v1
+        with:
+          notify_when: failure
+          status: failure # We do the filtering earlier
```

Comment:
Why do we?

Reply:
Because we have a complicated `if` statement instead of a simple "copy over the failure status from here"

This plugin was designed more for being used in a single task, not in a "run after everything else" finalizer.

Reply:
Basically, here you're supposed to put a variable substitution copying over the failure status from _somewhere_. Where would that be in this case? It can't be a single job because we care about multiple, and there's no catch-all "overall status" variable.